File: backend/app/schemas/export.py
"""Export schemas."""

from datetime import datetime
from typing import Optional, Any
from pydantic import BaseModel, ConfigDict

from app.models.export import ExportFormat, ExportStatus


class ExportBase(BaseModel):
    """Base export schema."""

    format: ExportFormat


class ExportCreate(ExportBase):
    """Export creation schema."""

    dataset_id: Optional[int] = None
    filters: Optional[dict[str, Any]] = None
    options: Optional[dict[str, Any]] = None


class ExportResponse(ExportBase):
    """Export response schema."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    dataset_id: Optional[int] = None
    status: ExportStatus
    file_path: Optional[str] = None
    file_size: Optional[int] = None
    record_count: Optional[int] = None
    download_url: Optional[str] = None
    expires_at: Optional[datetime] = None
    created_at: datetime
    completed_at: Optional[datetime] = None

File: frontend/src/types/export.ts
export enum ExportFormat {
  CSV = 'csv',
  EXCEL = 'excel',
  JSON = 'json',
  VCARD = 'vcard'
}

export enum ExportStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  COMPLETED = 'completed',
  FAILED = 'failed'
}

export interface Export {
  id: number
  user_id: number
  dataset_id?: number
  format: ExportFormat
  status: ExportStatus
  file_path?: string
  file_size?: number
  record_count?: number
  download_url?: string
  expires_at?: string
  created_at: string
  completed_at?: string
}
